src/libs/Performance.js

```diff
@@ -25,43 +33,157 @@ function diffObject(object, base) {
     return changes(object, base);
 }
 
-/**
- * Sets up an observer to capture events recorded in the native layer before the app fully initializes.
- */
-function setupPerformanceObserver() {
-    if (!canCapturePerformanceMetrics()) {
-        return;
-    }
+// When performance monitoring is disabled the implementations are blank
+/* eslint-disable import/no-mutable-exports */
+let setupPerformanceObserver = () => {};
+let printPerformanceMetrics = () => {};
+let markStart = () => {};
+let markEnd = () => {};
+let traceRender = () => {};
+let withRenderTrace = () => Component => Component;
+/* eslint-enable import/no-mutable-exports */
 
-    const performance = require('react-native-performance').default;
-    const PerformanceObserver = require('react-native-performance').PerformanceObserver;
-    new PerformanceObserver((list) => {
-        if (list.getEntries().find(entry => entry.name === 'nativeLaunchEnd')) {
-            performance.measure('nativeLaunch', 'nativeLaunchStart', 'nativeLaunchEnd');
-
-            // eslint-disable-next-line no-undef
-            if (__DEV__) {
-                performance.measure('jsBundleDownload', 'downloadStart', 'downloadEnd');
-            } else {
-                performance.measure('runJsBundle', 'runJsBundleStart', 'runJsBundleEnd');
-            }
-        }
-    }).observe({type: 'react-native-mark', buffered: true});
-}
+if (canCapturePerformanceMetrics()) {
+    /**
+     * Sets up an observer to capture events recorded in the native layer before the app fully initializes.
+     */
+    setupPerformanceObserver = () => {
+        const performanceReported = require('react-native-performance-flipper-reporter');
+        performanceReported.setupDefaultFlipperReporter();
 
-/**
- * Outputs performance stats. We alert these so that they are easy to access in release builds.
- */
-function printPerformanceMetrics() {
-    const performance = require('react-native-performance').default;
-    const entries = _.map(performance.getEntriesByType('measure'), entry => ({
-        name: entry.name, duration: Math.floor(entry.duration),
-    }));
-    alert(JSON.stringify(entries, null, 4));
+        const perfModule = require('react-native-performance');
+        perfModule.setResourceLoggingEnabled(true);
+        performance = perfModule.default;
+
+        // Monitor some native marks that we want to put on the timeline
+        new perfModule.PerformanceObserver((list, observer) => {
+            list.getEntries()
+                .forEach((entry) => {
```

Comment:
I think the observer is called only with new entries - the next callback call would not contain past entires
So if a mark like `runJsBundleEnd` repeats it doesn't matter if we use `find` or `foreach` it would still make us handle it twice
For these particular marks thought, this doesn't happen and it was easier to structure the logic